File: cloudtimes-ct/src/main/java/com/cloudtimes/account/mapper/CtAgentActivityMapper.java
package com.cloudtimes.account.mapper;

import java.util.List;
import com.cloudtimes.account.domain.CtAgentActivity;

/**
 * 代理活动Mapper接口
 * 
 * @author 沈兵
 * @date 2023-02-02
 */
public interface CtAgentActivityMapper 
{
    /**
     * 查询代理活动
     * 
     * @param id 代理活动主键
     * @return 代理活动
     */
    public CtAgentActivity selectCtAgentActivityById(String id);

    /**
     * 查询代理活动列表
     * 
     * @param ctAgentActivity 代理活动
     * @return 代理活动集合
     */
    public List<CtAgentActivity> selectCtAgentActivityList(CtAgentActivity ctAgentActivity);

    /**
     * 新增代理活动
     * 
     * @param ctAgentActivity 代理活动
     * @return 结果
     */
    public int insertCtAgentActivity(CtAgentActivity ctAgentActivity);

    /**
     * 修改代理活动
     * 
     * @param ctAgentActivity 代理活动
     * @return 结果
     */
    public int updateCtAgentActivity(CtAgentActivity ctAgentActivity);

    /**
     * 删除代理活动
     * 
     * @param id 代理活动主键
     * @return 结果
     */
    public int deleteCtAgentActivityById(String id);

    /**
     * 批量删除代理活动
     * 
     * @param ids 需要删除的数据主键集合
     * @return 结果
     */
    public int deleteCtAgentActivityByIds(String[] ids);
}

File: cloudtimes-ct/src/main/java/com/cloudtimes/account/mapper/CtAgentActivityMapper.kt
package com.cloudtimes.account.mapper

import com.cloudtimes.account.domain.CtAgentActivity
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Result
import org.apache.ibatis.annotations.ResultMap
import org.apache.ibatis.annotations.Results
import org.apache.ibatis.annotations.SelectProvider
import org.apache.ibatis.type.JdbcType
import org.mybatis.dynamic.sql.util.SqlProviderAdapter
import org.mybatis.dynamic.sql.select.render.SelectStatementProvider
import org.mybatis.dynamic.sql.util.mybatis3.CommonCountMapper
import org.mybatis.dynamic.sql.util.mybatis3.CommonDeleteMapper
import org.mybatis.dynamic.sql.util.mybatis3.CommonInsertMapper
import org.mybatis.dynamic.sql.util.mybatis3.CommonUpdateMapper

@Mapper
interface CtAgentActivityMapper : CommonCountMapper, CommonDeleteMapper, CommonInsertMapper<CtAgentActivity>, CommonUpdateMapper {
    @SelectProvider(type=SqlProviderAdapter::class, method="select")
    @Results(id="CtAgentActivityResult", value = [
        Result(column="user_id", property="userId", jdbcType=JdbcType.OTHER, id=true),
        Result(column="activity_id", property="activityId", jdbcType=JdbcType.OTHER)
    ])
    fun selectMany(selectStatement: SelectStatementProvider): List<CtAgentActivity>

    @SelectProvider(type=SqlProviderAdapter::class, method="select")
    @ResultMap("CtAgentActivityResult")
    fun selectOne(selectStatement: SelectStatementProvider): CtAgentActivity?
}
